### PR TITLE
[#5756] Bug Fix : Warehouse parameter systematically required 

### PR DESCRIPTION
**Pull request by fsalhi2**
### What changes were proposed in this pull request?

Removed the systematic validations in Iceberg Config and modified the instantiation of the warehouse attribute and uri attribute in the IcebergCatalogWrapper to conform with the new possibility (REST).

### Why are the changes needed?

The bug was blocking the creation of a rest catalog.

Fix: [#5756](https://github.com/apache/gravitino/issues/5756)

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

./gradlew build && compileDistribution && assembleDistribution + creation of the docker image there : https://hub.docker.com/r/fsalhi2/gravitino

Started gravitino with such configs : 

```
### Gravitino General Settings

gravitino.auxService.names = iceberg-rest
gravitino.iceberg-rest.classpath = iceberg-rest-server/libs, iceberg-rest-server/conf

### HTTP Server

gravitino.iceberg-rest.host = 0.0.0.0
gravitino.iceberg-rest.httpPort = 9001

### Storage

gravitino.iceberg-rest.io-impl = org.apache.iceberg.aws.s3.S3FileIO
gravitino.iceberg-rest.s3-access-key-id = XXXXX
gravitino.iceberg-rest.s3-secret-access-key = XXXXXX
gravitino.iceberg-rest.s3-path-style-access = true
gravitino.iceberg-rest.s3-endpoint = http://minio:9000/
gravitino.iceberg-rest.s3-region = us-east-1

### JDBC

gravitino.iceberg-rest.catalog-backend = jdbc
gravitino.iceberg-rest.uri = jdbc:mysql://mysql:3306/
gravitino.iceberg-rest.warehouse = s3://lake/catalog
gravitino.iceberg-rest.jdbc.user = root
gravitino.iceberg-rest.jdbc.password = XXXXXX
gravitino.iceberg-rest.jdbc-driver = com.mysql.cj.jdbc.Driver
```

Was able to create a catalog through Web UI and start working on the scheme.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -20,18 +20,18 @@
 org.gradle.parallel=true
```
[the remark this answers is not in the document]
Comment (fsalhi2) [reply gradle.properties:20]: Yes the changes in gradle.properties aren't related , i've forgot to disable them in my commits.

File: iceberg/iceberg-common/src/main/java/org/apache/gravitino/iceberg/common/ops/IcebergCatalogWrapper.java [hunk 2]
```diff
@@ -82,9 +82,12 @@ public IcebergCatalogWrapper(IcebergConfig icebergConfig) {
     this.catalogBackend =
         IcebergCatalogBackend.valueOf(
             icebergConfig.get(IcebergConfig.CATALOG_BACKEND).toUpperCase(Locale.ROOT));
-    if (!IcebergCatalogBackend.MEMORY.equals(catalogBackend)) {
+    if (!IcebergCatalogBackend.MEMORY.equals(catalogBackend)
+        && !IcebergCatalogBackend.REST.equals(catalogBackend)) {
       // check whether IcebergConfig.CATALOG_WAREHOUSE exists
       icebergConfig.get(IcebergConfig.CATALOG_WAREHOUSE);
```
[the remark this answers is not in the document]
Comment (fsalhi2) [reply iceberg/iceberg-common/src/main/java/org/apache/gravitino/iceberg/common/ops/IcebergCatalogWrapper.java:88]: I am looking into this in some moments. Thanks for your feedback

File: catalogs/catalog-lakehouse-iceberg/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java [hunk 3]
```diff
@@ -74,10 +74,11 @@ public class IcebergCatalogPropertiesMetadata extends BaseCatalogPropertiesMetad
                 false /* reserved */),
             stringRequiredPropertyEntry(
                 URI, "Iceberg catalog uri config", false /* immutable */, false /* hidden */),
-            stringRequiredPropertyEntry(
+            stringOptionalPropertyEntry(
                 WAREHOUSE,
                 "Iceberg catalog warehouse config",
                 false /* immutable */,
+                null, /* defaultValue */
```
Comment (tengqm) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java:81]: We could have prevented this kind of obvious errors if we apply a stricter coding style.
Can you help also wrap line 76?

Comment (fsalhi2) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java:81]: Yup

**Comment by fsalhi2** [comment]
@FANNG1 @tengqm The changes have been done.

**Comment by tengqm** [comment]
lgtm

File: catalogs/catalog-lakehouse-iceberg/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java (hunk 3 above)
Comment (tengqm) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergCatalogPropertiesMetadata.java:81]: Looks like that this is not updated yet?

File: gradle.properties [hunk 4]
```diff
@@ -23,7 +23,7 @@ org.gradle.caching=true
 org.gradle.jvmargs=-Xmx4g
 
 # version that is going to be updated automatically by releases
-version = 0.8.0-incubating-SNAPSHOT
+version = 0.7.0-incubating-SNAPSHOT
```
Comment (FANNG1) [inline gradle.properties:26]: please don't update these properties

**Comment by FANNG1** [comment]
LGTM except one comments

File: gradle.properties (hunk 4 above)
Comment (fsalhi2) [reply gradle.properties]: This is fixed now, thanks for the feedback

Comment (FANNG1) [reply gradle.properties]: revert pythonVersion too?

Comment (fsalhi2) [reply gradle.properties]: oops sorry, that's done

**Review by FANNG1 — APPROVED** [review]

**Comment by jerryshao** [comment]
Merging to main branch, thanks @fsalhi2 for your contribution.

**Pull request merged**